### PR TITLE
log optimization modification: adjust log level, remove duplicate log

### PR DESCRIPTION
**Pull request by gy95**
Signed-off-by: gy95 <guoyao17@huawei.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
1. adjust some log level: from info to error, from info to debug.
2. remove duplicate error log print if out caller also print error logs, inner function dont need to also print logs

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go [hunk 1]
```diff
@@ -79,7 +79,6 @@ func Receive(module string) (model.Message, error) {
 	if err == nil {
 		return message, nil
 	}
-	klog.Warningf("Receive: failed to receive message, error:%v", err)
 	return message, err
```
Comment (zc2638) [inline staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go:82]: Looks like it can be merged.

Comment (fisherxu) [inline staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go:82]:
```suggestion
	return message, error.new("Receive: failed to receive message, error:%v", err) 
```
?

Comment (zc2638) [reply staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go:82]: No judgment `err = = nil`

Comment (gy95) [reply staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go]: done.

File: edge/pkg/edged/edged.go [hunk 2]
```diff
@@ -1015,19 +1015,17 @@ func (e *edged) podRemoveWorkerRun(consumers int) {
...
 	podName := namespacedName.Name
-	klog.Infof("start to consume added pod [%s]", podName)
+	klog.V(4).Infof("start to consume added pod [%s]", podName)
 	pod, ok := e.podManager.GetPodByName(namespacedName.Namespace, podName)
 	if !ok || pod.DeletionTimestamp != nil {
 		return apis.ErrPodNotFound
 	}
 
 	if err := e.makePodDataDirs(pod); err != nil {
-		klog.Errorf("Unable to make pod data directories for pod %q: %v", format.Pod(pod), err)
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:1025]: Do we need to return all the info in `klog.Errorf`?

Comment (gy95) [reply edge/pkg/edged/edged.go:1025]: ok

File: edge/pkg/edged/status/status_manager.go [hunk 3]
```diff
@@ -112,8 +112,9 @@ func (m *manager) updatePodStatus() {
 		err := m.metaClient.PodStatus(pod.Namespace).Update(pod.Name, edgeapi.PodStatusRequest{UID: pod.UID, Name: pod.Name, Status: s})
 		if err != nil {
 			klog.Errorf("Update pod status failed err :%v", err)
+		} else {
+			klog.Infof("Status for pod %s updated successfully", pod.Name)
```
Comment (fisherxu) [inline edge/pkg/edged/status/status_manager.go:116]:
```suggestion
			klog.V(4).Infof("Status for pod %s updated successfully", pod.Name)
```

File: edge/pkg/edged/edged.go [hunk 4]
```diff
@@ -1043,8 +1041,7 @@ func (e *edged) consumePodAddition(namespacedName *types.NamespacedName) error {
 	}
 	curPodStatus, err := e.podCache.GetNewerThan(podUID, t.(time.Time))
 	if err != nil {
-		klog.Errorf("pod %s cache newer failed: %v", podName, err)
-		return err
+		return fmt.Errorf("pod %s cache newer failed: %v", podName, err)
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:1044]: Seems that the err will not be printed in calling func, please check all the errors modified.

File: edge/pkg/edged/status/status_manager.go (hunk 3 above)
Comment (gy95) [reply edge/pkg/edged/status/status_manager.go]: done

**Comment by gy95** [comment]
PTAL 😄  @fisherxu @zc2638

File: edge/pkg/edged/status/status_manager.go [hunk 5]
```diff
@@ -111,8 +111,9 @@ func (m *manager) updatePodStatus() {
 		err := m.metaClient.PodStatus(pod.Namespace).Update(pod.Name, edgeapi.PodStatusRequest{UID: pod.UID, Name: pod.Name, Status: s})
 		if err != nil {
 			klog.Errorf("Update pod status failed err :%v", err)
```
Comment (fisherxu) [inline edge/pkg/edged/status/status_manager.go:113]: We should return here instead of adding the else. :)

File: edge/pkg/edged/edged.go [hunk 6]
```diff
@@ -1009,25 +1009,23 @@ func (e *edged) podRemoveWorkerRun(consumers int) {
 
 func (e *edged) consumePodAddition(namespacedName *types.NamespacedName) error {
 	podName := namespacedName.Name
-	klog.Infof("start to consume added pod [%s]", podName)
+	klog.V(4).Infof("start to consume added pod [%s]", podName)
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:1012]: We can remain the klog.Infof here, always need to check whether the pods are handled by edged.

File: edge/pkg/edged/edged.go [hunk 7]
```diff
@@ -1048,30 +1045,29 @@ func (e *edged) consumePodAddition(namespacedName *types.NamespacedName) error {
...
+				return fmt.Errorf("sync pod failed: %v", err)
 			}
 		}
 
 		return nil
 	}
 
 	e.workQueue.Enqueue(pod.UID, utilwait.Jitter(time.Minute, workerResyncIntervalJitterFactor))
-	klog.Infof("consume added pod [%s] successfully\n", podName)
+	klog.V(4).Infof("consume added pod [%s] successfully\n", podName)
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:1056]: ditto

File: edge/pkg/edged/edged.go [hunk 8]
```diff
@@ -1048,30 +1045,29 @@ func (e *edged) consumePodAddition(namespacedName *types.NamespacedName) error {
...
 	return nil
 }
 
 func (e *edged) consumePodDeletion(namespacedName *types.NamespacedName) error {
 	podName := namespacedName.Name
-	klog.Infof("start to consume removed pod [%s]", podName)
+	klog.V(4).Infof("start to consume removed pod [%s]", podName)
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:1062]: ditto

File: edge/pkg/edged/edged.go [hunk 9]
```diff
@@ -1082,7 +1078,7 @@ func (e *edged) consumePodDeletion(namespacedName *types.NamespacedName) error {
 		}
 		return fmt.Errorf("consume removed pod [%s] failed, %v", podName, err)
 	}
-	klog.Infof("consume removed pod [%s] successfully\n", podName)
+	klog.V(4).Infof("consume removed pod [%s] successfully\n", podName)
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:1081]: ditto

Comment (gy95) [reply edge/pkg/edged/edged.go]: done

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**